src/taskpane/index.tsx

```diff
@@ -1,38 +1,27 @@
-import App from "./components/App";
-import { AppContainer } from "react-hot-loader";
-import { initializeIcons } from "@fluentui/font-icons-mdl2";
-import { ThemeProvider } from "@fluentui/react";
 import * as React from "react";
-import * as ReactDOM from "react-dom";
+import { createRoot } from "react-dom/client";
+import App from "./components/App";
+import { FluentProvider, webLightTheme } from "@fluentui/react-components";
 
 /* global document, Office, module, require */
 
-initializeIcons();
-
-let isOfficeInitialized = false;
```

Comment:
I don't know the history of this, but ReactDom.render was only called immediately after this was set to true (in Office.onReady), so when the React App component rendered, the branch that ran when `isOfficeInitialized` was false NEVER ran in the taskpane. It would only run if the user pasted the add-in's URL into a browser window. But devs only do this to troubleshoot UI issues, so we don't want an alternate UI to appear in the browser.

Reply:
Now that you mention it, I believe you need this in order for the page to render in the browser at all.  Office.js check where it's being loaded and won't load outside of an Office host.  So I think if you were to open the page in the browser it wouldn't load at all because Office.js throws and error and the JavaScript doesn't complete.

Reply:
I tested that. It opens just fine in the browser. This is generally true of all add-ins. If opened in a browser there is an error about Office, but the UI does open and render.

Reply:
OK . . . looks like the office.js load just gives a warning and it load.  I guess this is OK then.